src/utils/paginationHelper.js

```diff
@@ -0,0 +1,63 @@
+export function calculatePages(numPreviousPages, numNextPages, currentPage, totalPages) {
+  const paginationElements = [
+    {
+      displayText: currentPage,
+      requiresLink: false,
+      isCurrent: true
+    }
+  ]
+
+  const ellipsis = {
+    displayText: '...',
+    requiresLink: false
+  }
+
+  for (let i = 1; i <= numPreviousPages; i++) {
+    if (currentPage - i < 1) {
+      break;
+    }
+
+    paginationElements.unshift({
+      displayText: currentPage - i,
+      requiresLink: true,
+    })
+  }
+
+  const firstElementText = paginationElements[0].displayText
+  if (firstElementText > 1) {
+    if (firstElementText > 2) {
+      paginationElements.unshift(ellipsis)
+    }
+
+    paginationElements.unshift({
+      displayText: 1,
+      requiresLink: true
+    })
+  }
+
+  for (let j = 1; j <= numNextPages; j++) {
```

Comment:
You can use `i` again since `let` does not make the variable visible in the parent scope